src/editor/codemirror/CodeMirror.tsx

```diff
@@ -50,12 +55,21 @@ const CodeMirror = ({
   codeStructureSettings,
 }: CodeMirrorProps) => {
   // Really simple model for now as we only have one editor at a time.
-  const [, setActiveEditor] = useActiveEditor();
+  const [, setActiveEditor] = useActiveEditorActionsState();
   const uri = createUri(selection.file);
   const elementRef = useRef<HTMLDivElement | null>(null);
   const viewRef = useRef<EditorView | null>(null);
   const client = useLanguageServerClient();
   const intl = useIntl();
+  const [, setEditorInfo] = useActiveEditorInfoState();
+
+  // Reset undo/redo events on file change.
+  useEffect(() => {
+    setEditorInfo({
+      undo: 0,
+      redo: 0,
+    });
+  }, [setEditorInfo]);
```

Comment:
Tempting to deal with this elsewhere.

Reply:
Tried this but it's a faff because of the notifier being a facet we need to configure up front. Not worth it absent a better idea.